app/assets/javascripts/controllers/security_group/security_group_form_controller.js

```diff
@@ -1,54 +1,100 @@
-  $scope.securityGroupModel = { name: '' };
-  $scope.formId = securityGroupFormId;
-  $scope.afterGet = false;
-  $scope.modelCopy = angular.copy( $scope.securityGroupModel );
-  $scope.model = "securityGroupModel";
-
-  ManageIQ.angular.scope = $scope;
-
-  if (securityGroupFormId == 'new') {
-    $scope.securityGroupModel.name = "";
-    $scope.securityGroupModel.description = "";
-    $scope.newRecord = true;
-  } else {
-    miqService.sparkleOn();
-    API.get("/api/security_groups/" + securityGroupFormId + "?attributes=cloud_tenant").then(function(data) {
-      $scope.afterGet = true;
-      $scope.securityGroupModel.name = data.name;
-      $scope.securityGroupModel.description = data.description;
-      $scope.securityGroupModel.cloud_tenant_name = data.cloud_tenant.name;
-      $scope.modelCopy = angular.copy( $scope.securityGroupModel );
+ManageIQ.angular.app.controller('securityGroupFormController', ['securityGroupFormId', 'miqService', 'API', function(securityGroupFormId, miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.securityGroupModel = {
+      name: "",
+      description: "",
+      cloud_tenant_name: "",
+      firewall_rules: [],
+    };
+
+    getSecurityGroups();
+    vm.hostProtocols = ["", "TCP", "UDP", "ICMP"];
+    vm.networkProtocols = ["IPV4", "IPV6"];
+    vm.directions = ["inbound", "outbound"];
+
+    vm.formId = securityGroupFormId;
+    vm.model = "securityGroupModel";
+    vm.newRecord = securityGroupFormId == "new";
+    vm.afterGet = false;
+    vm.modelCopy = angular.copy( vm.securityGroupModel );
+    vm.saveable = miqService.saveable;
+    ManageIQ.angular.scope = vm;
+
+    if (vm.newRecord) {
+      vm.securityGroupModel.name = "";
+    } else {
+      miqService.sparkleOn();
+      API.get("/api/security_groups/" + securityGroupFormId + "?attributes=cloud_tenant,firewall_rules").then(function(data) {
+        vm.securityGroupModel.name = data.name;
+        vm.securityGroupModel.description = data.description;
+        vm.securityGroupModel.cloud_tenant_name = data.cloud_tenant.name;
+        vm.securityGroupModel.firewall_rules = data.firewall_rules;
+        vm.afterGet = true;
+        vm.modelCopy = angular.copy( vm.securityGroupModel );
+      }).catch(miqService.handleFailure);
       miqService.sparkleOff();
+    }
+  };
+
+  var getSecurityGroups = function() {
+    API.get("/api/security_groups/?expand=resources&attributes=ems_ref,id,name").then(function(data) {
+      vm.security_groups_list = data.resources;
     }).catch(miqService.handleFailure);
-  }
+  };
 
-  $scope.addClicked = function() {
+  vm.addClicked = function() {
     var url = 'create/new' + '?button=add';
-    miqService.miqAjaxButton(url, $scope.securityGroupModel, { complete: false });
+    miqService.miqAjaxButton(url, vm.securityGroupModel, { complete: false });
```

Comment:
Would it be possible to use a `API.post` call here to add a new SecurityGroup record?
And also below for saving/updating an existing SecurityGroup record?